Core/Core/SwiftUIViews/ListCellView/ListCellViewModel.swift

```diff
@@ -0,0 +1,142 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+class ListCellViewModel: ObservableObject {
+
+    let cellStyle: ListCellView.ListCellStyle
+    let title: String
+    let subtitle: String?
+    let error: String?
+    let selectionState: ListCellView.SelectionState
+    let isCollapsed: Bool?
+    let selectionDidToggle: (() -> Void)?
+    let collapseDidToggle: (() -> Void)?
+    let removeItemPressed: (() -> Void)?
+    let progress: Float?
```

Comment:
We can now have error and progress at the same time, wouldn't it make sense to have a state property instead?

Reply:
This tends to be a generic item, so the plan is to further delegate such logic to the screen's viewmodel. It's also a feasible scenario where we show a progress state at where the transfer has failed.